systemd: ProtectHome was too strict

rumpelsepp opened the pull request:
"ProtectHome=true" was too strict and caused problems when running under
the system instance. The problem does not appear when syncthing-inotify
is running under the user instance, thus I missed it in the first place.

"ProtectHome=read-only" fixes the problem.

Reported by: madalu mdl@imapmail.org
Reported on: https://aur.archlinux.org/packages/syncthing-inotify/

rumpelsepp committed:
systemd: ProtectHome was too strict

"ProtectHome=true" was too strict and caused problems when running under
the system instance. The problem does not appear when syncthing-inotify
is running under the user instance, thus I missed it in the first place.

"ProtectHome=read-only" fixes the problem.

Reported by: madalu <mdl@imapmail.org>
Reported on: https://aur.archlinux.org/packages/syncthing-inotify/

Zillode commented:
Does this actually allow inotify to watch dirs outside the home dir?

rumpelsepp commented:
Yes.

* `ProtectHome=read-only` mounts `/home` and `/run/user` read-only and
* `ProtectSystem=full` mounts `/usr/, `/boot` and `etc/` read-only

for the `syncthing-inotify.service`. I only see a problem with `/tmp`, because of `PrivateTmp=true` which provides a seperate `/tmp` instance for this service. Basically it looks like this:

```
.
├── systemd-private-78e6d6e2455943d3ad5a88d1a108c1df-colord.service-JrjCHm
│   └── tmp
├── systemd-private-78e6d6e2455943d3ad5a88d1a108c1df-syncthing-inotify.service-qtTGNm
│   └── tmp
├── systemd-private-78e6d6e2455943d3ad5a88d1a108c1df-systemd-timesyncd.service-JOPsHb
│   └── tmp
```

Zillode commented:
I think we should drop the ```PrivateTmp``` option because inotify does not write temp files anyway. If someone wants to sync the ```/tmp``` folder he could be caught be surprise, don't you agree?

rumpelsepp commented:
I think we should drop it. If sb decides to watch `/tmp` odd things will happen…

> because inotify does not write temp files anyway

When it does not write to `/tmp` anyway we can safely remove it.

rumpelsepp committed:
systemd: "PrivateTmp" is too strict

We decided to drop "PrivateTmp" because syncthing-inotify does not write
to "/tmp". Furthermore if somebody decides to watch "/tmp" odd things
will happen as "PrivateTmp" mounts an isolated "/tmp" instance for
the service.

Zillode commented:
Could you squash this please?:) Thanks!
